specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,268 @@
+                            {
+                                MessageBox(
+                                    nullptr, exceptionMessage.get(), L"ExecuteScript Result", MB_OK);
+                            }
+
+                            // Get the exception detail, it's a json struct data with all exception infomation
+                            // , we can parse it and get the detail what we need.
+                            wil::unique_cotaskmem_string exceptionDetail;
+                            if (exception && exception->get_Detail(&exceptionDetail) == S_OK)
+                            {
+                                MessageBox(
+                                    nullptr, exceptionDetail.get(), L"ExecuteScript Result", MB_OK);
+                            }
+                        }
+                        else
+                        {
+                            MessageBox(
+                                nullptr, L"Get exception failed", L"ExecuteScript Result", MB_OK); 
+                        }
+                    }
+
+                }
+            }
+        )
+    )
+}
+
+```
+## .NET and WinRT
+
+```c#
+void ExecuteScriptWithResultAsync(String script)
+{
+    var environment = webView2Control.CoreWebView2.Environment;
+    CoreWebView2ExecuteScriptResult result = await ExecuteScriptWithResultAsync(script);
+
+    bool isSuccess = result.IsSuccess;
+    if (isSuccess) 
+    {
+        Debug.WriteLine($"execute script success.");
+
+        Debug.WriteLine($"json result received: {result.ResultAsJson}");
+        try 
+        {
+            string stringResult = result.TryGetResultAsString();
+            Debug.WriteLine($"get string result success: {stringResult}");
+        } 
+        catch (ArgumentException) 
+        {
+            Debug.WriteLine($"Non-string message received");
+        }
+    } 
+    else 
+    {
+        Debug.WriteLine($"execute script failed.");
+        CoreWebView2ExecuteScriptException exception = result.Exception;
+        Debug.WriteLine($"exception name: {exception.Name}");
+        Debug.WriteLine($"exception message: {exception.Message}");
+        Debug.WriteLine($"exception Detail: {exception.Detail}");
+    }
+}
+```
+
+# API Details
+## Win32 C++
+```c++
+/// This is the exception struct when ExecuteScriptWithResult return false, user can
+/// use get_Exception to get it.
+[uuid(82F22B72-1B22-403E-A0B9-A8816C9C8E45), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptException : IUnknown {
+
+  /// This will return the exception className, it would be got from the 
+  /// `result.exceptionDetail.exception.className` in json result, this
+  /// could be empty if the exception doesn't have the specified element, 
+  /// such as user active throw an exception like `throw "abc"`.
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+
+  /// This will return the exception message, it would be got from the 
```

Comment:
I'd recommend changing these lines 179 through 182 to
```
The Message is the exception's message and potentially stack. In the JSON it is exceptionDetail.exception.description. This is the empty string if the exception doesn't have a description. This can happen if the script throws a non-Error object such as throw "abc";.
```